apple-codesign/src/cli/certificate_source.rs

```diff
@@ -263,6 +268,79 @@ impl KeySource for MacosKeychainSigningKey {
     }
 }
 
+#[derive(Args, Clone, Debug, Deserialize, Eq, PartialEq)]
+pub struct WindowsStoreSigningKey {
+    /// (Windows only) Windows Store to operate on
+    #[arg(long = "store-name", value_parser = crate::cli::STORE_NAMES, value_name = "STORE")]
+    pub stores: Vec<String>,
+
+    /// (Windows only) SHA-1 fingerprint of certificate in Windows Store to use
+    #[arg(
+        long = "store-cert-fingerprint",
+        value_name = "SHA1 FINGERPRINT"
+    )]
+    pub sha1_fingerprint: Option<String>,
+}
+
+impl KeySource for WindowsStoreSigningKey {
+    #[cfg(target_os = "windows")]
+    fn resolve_certificates(&self) -> Result<SigningCertificates, AppleCodesignError> {
+        // No arguments pertinent to store. Don't even speak to the
+        // Windows API since this could only error.
+        if self.stores.is_empty() && self.sha1_fingerprint.is_none() {
+            return Ok(Default::default());
+        }
+
+        // Collect all the store names to search.
+        let stores = if self.stores.is_empty() {
+            vec!["user".to_string()]
+        } else {
+            self.stores.clone()
+        };
+
+        let stores = stores
+            .into_iter()
+            .map(|store| {
+                StoreName::try_from(store.as_str())
+                    .expect("clap should have validated store name values")
+            })
+            .collect::<Vec<_>>();
+
+        // Now iterate all the keychains and try to find requested certificates.
+        let mut res = SigningCertificates::default();
+
+        for store in stores {
+            for cert in windows_store_find_code_signing_certificates(store, StoreType::MY)? {
```

Comment:
Should `StoreType::MY` be hardcoded here? Is there any benefit to allowing it to be controlled by an end-user?

Reply:
That makes sense, as end user certificates are normally located in the `MY` store.